File: SquareDigit.cpp
#include <iostream>
#include <cmath>
#include <chrono>
#include <ctime>
long long squareDigits(long num);

int main()
{
    long sample;

    std::cout << "Enter a number: ";
    std::cin >> sample;
    std::cin.ignore();

    auto start = std::chrono::system_clock::now();
    std::cout << sample << " = " << squareDigits(sample) << std::endl;
    auto end = std::chrono::system_clock::now();

    std::chrono::duration<double> elapsedSeconds = end - start;
    std::cout << "Time Elapsed: " << elapsedSeconds.count() << "ms." << std::endl;
    return 0;
}

long long squareDigits(long num)
{
    std::string numString = std::to_string(num);
    std::string result = "";

    for (int i = 0; i < numString.length(); i++)
        result.append(std::to_string((int)pow(numString.at(i) - '0', 2))); // character to integer

    //std::cout << result << std::endl;
    return std::stoll(result);
}

File: Java/SquareDigit.java
import java.util.stream.Collectors;

/**
 * Welcome. In this kata, you are asked to square every digit of a number. For
 * example, if we run 9119 through the function, 811181 will come out, because
 * 9^2 is 81 and 1^2 is 1.
 * 
 * Note: The function accepts an integer and returns an integer
 */

public class SquareDigit {

    static int squareDigits(int num) {
        char[] digits = String.valueOf(num).toCharArray();
        String result = "";

        for (char digit : digits)
            result += Integer.toString((int) Math.pow(Character.getNumericValue(digit), 2));

        return Integer.parseInt(result);
    }

    // Other answers
    static int squareDigits2(int n) {
        return Integer.parseInt(String.valueOf(n).chars().map(x -> x - '0').map(i -> i * i).mapToObj(String::valueOf)
                .collect(Collectors.joining("")));
    }

    public static void main(String[] args) {

        int sample = 9119;

        long start = System.currentTimeMillis();
        System.out.println("Result: " + squareDigits(sample));
        long end = System.currentTimeMillis();
        System.out.println("Time Elapsed: " + ((end - start) / 1000000) + " ms");

        start = System.currentTimeMillis();
        System.out.println("Result: " + squareDigits2(sample));
        end = System.currentTimeMillis();
        System.out.println("Time Elapsed: " + ((end - start) / 1000000) + " ms");
    }
}